Rubberduck.Core/AutoComplete/AutoCompleteService.cs

```diff
@@ -24,16 +18,21 @@ public class AutoCompleteService : IDisposable
             new SelfClosingPair('{', '}'),
         };
 
+        private readonly AutoCompleteKeyDownHandler _handler;
         private readonly SelfClosingPairCompletionService _selfClosingPairCompletion;
+        private readonly ICodeStringPrettifier _prettifier;
 
         private AutoCompleteSettings _settings;
         private bool _popupShown;
-        private bool _enabled = false;
+        private bool _enabled;
         private bool _initialized;
 
-        public AutoCompleteService(IGeneralConfigService configService, SelfClosingPairCompletionService selfClosingPairCompletion)
+        public AutoCompleteService(IGeneralConfigService configService, SelfClosingPairCompletionService selfClosingPairCompletion, ICodeStringPrettifier prettifier)
         {
+            _handler = new AutoCompleteKeyDownHandler(()=>_settings, ()=>_selfClosingPairs, ()=>_selfClosingPairCompletion, ()=>_prettifier);
```

Comment:
todo: find a way to inject this

Reply:
taking shape now, I'll have the smart-concat handling in its own class tomorrow, with a bunch of tests, some of which I'm expecting to fail at this time, given the little testing I did in the VBE.